### PR TITLE
[Do_while] Fix pipeline component with do-while operator cannot be submitted.

### PR DESCRIPTION
**Pull request by lalala123123**
# Description
1. change error message to "The max iteration count cannot be less than 0 or larger than 1000"
2.  Remove the null or empty check for dowhile settings
3. Fix raise error when dowhile node in sub_pipeline

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/tests/recordings/pipeline_job/e2etests/test_pipeline_job.pyTestConditionalNodeInPipelinetest_pipeline_with_invalid_do_while_node.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  "Entries": [],
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/tests/recordings/pipeline_job/e2etests/test_pipeline_job.pyTestConditionalNodeInPipelinetest_pipeline_with_invalid_do_while_node.json:2]: This seems like an UT?

File: sdk/ml/azure-ai-ml/tests/pipeline_job/e2etests/test_pipeline_job.py [hunk 2]
```diff
@@ -1629,3 +1581,70 @@ def test_reused_pipeline_child_job_download(
         assert next(artifact_dir.iterdir(), None), "No artifacts were downloaded"
         assert output_dir.exists()
         assert next(output_dir.iterdir(), None), "No outputs were downloaded"
+
+
+@pytest.mark.usefixtures(
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/tests/pipeline_job/e2etests/test_pipeline_job.py:1586]: maybe split this into a new test file

File: sdk/ml/azure-ai-ml/tests/recordings/pipeline_job/e2etests/test_pipeline_job.pyTestConditionalNodeInPipelinetest_pipeline_with_invalid_do_while_node.json (hunk 1 above)
Comment (lalala123123) [reply sdk/ml/azure-ai-ml/tests/recordings/pipeline_job/e2etests/test_pipeline_job.pyTestConditionalNodeInPipelinetest_pipeline_with_invalid_do_while_node.json]: move it to unittests

File: sdk/ml/azure-ai-ml/tests/pipeline_job/e2etests/test_pipeline_job.py (hunk 2 above)
Comment (lalala123123) [reply sdk/ml/azure-ai-ml/tests/pipeline_job/e2etests/test_pipeline_job.py]: split these test cases into a new test file

**Review by wangchao1230 — APPROVED** [review]

**Pull request merged**